src/core/positioning/tcpreceiver.cpp

```diff
@@ -22,8 +22,10 @@ TcpReceiver::TcpReceiver( const QString &address, const int port, QObject *paren
   , mPort( port )
   , mSocket( new QTcpSocket() )
 {
-  connect( mSocket, &QAbstractSocket::stateChanged, this, &TcpReceiver::setSocketState );
   connect( mSocket, qOverload<QAbstractSocket::SocketError>( &QAbstractSocket::errorOccurred ), this, &TcpReceiver::handleError );
+  connect( mSocket, &QTcpSocket::stateChanged, this, [=]( QAbstractSocket::SocketState state ) {
```

Comment:
Again here I think you can just connect to the abstract gnss receiver class' function.